src/lib/OS/SyscallHandler.cc

```diff
@@ -526,7 +526,41 @@ void SyscallHandler::handleSyscall() {
       break;
     }
     case 179: {  // sysinfo
+      // Sysinfo populates a sysinfo struct containing system information.
+      // Currently, the majority of the sysinfo struct entries will be 0 for
+      // SimEng's emulation of the syscall, therefore, we begin with a zero'ed
+      // out std::array, `sysInfoVals`, to represent this struct. The mapping
+      // between struct and `sysInfoVals` entries is as followed:
+      //
+      // {uptime, loads[0], loads[1], loads[2], totalram,
+      // freeram, sharedram, bufferram, totalswap, freeswap,
+      // procs,totalhigh, freehigh, mem_unit}
+      //
+      // (https://man7.org/linux/man-pages/man2/sysinfo.2.html has been used to
+      // define the entries to be mapped and understand what they equate to.)
+      //
+      // Each entry is represented as an uint64_t value and later narrowed
+      // implicity within the RegisterValue constructor where necessary.
+      // `sysInfoValsSizes` is used to denote the sizes of each entry so that it
+      // occupies the correct amount of memory after the syscall is concluded.
+
+      uint64_t infoPtr = currentInfo_.registerArguments[0].get<uint64_t>();
       stateChange = {ChangeType::REPLACEMENT, {currentInfo_.ret}, {0ull}};
+      std::array<uint64_t, 14> sysInfoVals;
```

Comment:
Quick question: are you sure that `uint64_t` is initialised as 0? if not could we add:
`std::fill(sysInfoVals.begin(), sysInfoVals.end(), 0);`

Reply:
It may be impl defined so will add the fill